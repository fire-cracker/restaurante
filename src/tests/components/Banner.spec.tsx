import React from 'react'
import { render } from '@testing-library/react'

import Banner from '../../components/Banner'

describe('Banner', () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = render(<Banner />)
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  test('renders Banner component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('renders component text', () => {
    const { getByText } = wrapper
    const linkElement = getByText('Restaurante Exquisite Cuisine')
    const openingTimeElement = getByText('OPENING HOURS')
    expect(linkElement).toBeInTheDocument()
    expect(openingTimeElement).toBeInTheDocument()
  })
})
