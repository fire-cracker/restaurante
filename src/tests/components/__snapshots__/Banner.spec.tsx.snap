// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders Banner component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="p-0 jumbotron jumbotron-fluid"
      >
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="embed-responsive embed-responsive-1by1"
            >
              <div
                class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
              >
                <div
                  class="text-0 text-darkkhaki"
                >
                  <p>
                    Restaurante Exquisite Cuisine
                  </p>
                </div>
                <div
                  class="text-1 text-white"
                >
                  <p
                    class="font-weight-normal"
                  >
                    Enjoy our Food Experience
                  </p>
                </div>
                <div
                  class="divider text-darkkhaki d-flex flex-row"
                >
                  <p
                    class="line my-2 mx-2"
                  >
                     
                  </p>
                  <p
                    class="icon"
                  >
                    <i
                      class="fa fa-cutlery"
                    />
                  </p>
                  <p
                    class="line my-2 mx-2"
                  >
                     
                  </p>
                </div>
                <div
                  class="text-2 text-white"
                >
                  <p>
                    OPENING HOURS
                  </p>
                </div>
                <div
                  class="text-3 text-white d-flex flex-row"
                >
                  <p
                    class="days"
                  >
                    Monday - Friday: 
                  </p>
                  <p
                    class="time text-dimgray"
                  >
                     07:30AM - 23:00PM
                  </p>
                </div>
                <div
                  class="text-4 text-white d-flex flex-row"
                >
                  <p
                    class="days"
                  >
                    Weekends: 
                  </p>
                  <p
                    class="time text-dimgray"
                  >
                     07:30AM - 03:00PM
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="p-0 jumbotron jumbotron-fluid"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="embed-responsive embed-responsive-1by1"
          >
            <div
              class="banner d-flex flex-column justify-content-center align-items-center embed-responsive-item col"
            >
              <div
                class="text-0 text-darkkhaki"
              >
                <p>
                  Restaurante Exquisite Cuisine
                </p>
              </div>
              <div
                class="text-1 text-white"
              >
                <p
                  class="font-weight-normal"
                >
                  Enjoy our Food Experience
                </p>
              </div>
              <div
                class="divider text-darkkhaki d-flex flex-row"
              >
                <p
                  class="line my-2 mx-2"
                >
                   
                </p>
                <p
                  class="icon"
                >
                  <i
                    class="fa fa-cutlery"
                  />
                </p>
                <p
                  class="line my-2 mx-2"
                >
                   
                </p>
              </div>
              <div
                class="text-2 text-white"
              >
                <p>
                  OPENING HOURS
                </p>
              </div>
              <div
                class="text-3 text-white d-flex flex-row"
              >
                <p
                  class="days"
                >
                  Monday - Friday: 
                </p>
                <p
                  class="time text-dimgray"
                >
                   07:30AM - 23:00PM
                </p>
              </div>
              <div
                class="text-4 text-white d-flex flex-row"
              >
                <p
                  class="days"
                >
                  Weekends: 
                </p>
                <p
                  class="time text-dimgray"
                >
                   07:30AM - 03:00PM
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
